source/slang/slang-emit.cpp

```diff
@@ -845,13 +845,14 @@ Result linkAndOptimizeIR(
     {
     case CodeGenTarget::HLSL:
         {
-            legalizeNonStructParameterToStructAndBackHLSL(irModule);
+            legalizeNonStructParameterToStructHLSL(irModule);
```

Comment:
If I look down a few lines, the naming patter has "For".
You may want to use "XXXForHLSL" for consistency.
```
legalizeImageSubscriptForGLSL(irModule);
legalizeConstantBufferLoadForGLSL(irModule);
legalizeDispatchMeshPayloadForGLSL(irModule);
```